woocommerce/woocommerce-gutenberg-products-block
# Migrate interactivity stock filter to new store API, add improvements and bugfixes

## What

This is an update to the experimental interactivity stock filter block, that can be used with experimental collection filter block.

It adds [suggestions](https://github.com/woocommerce/woocommerce-blocks/pull/11663#discussion_r1391953722) made by @dinhtungdu :

1. Fix a bug where child click handler bubbled up to parent, selecting item caused `toggleIsOpen` to fire
2. Enhance the Dropdown to support unselecting an item
3. When generating the Dropdown and Checklist options, include the stock count only if `showCounts` is true.

Also while this PR was WIP the new store API code was merged, so this has now been updated to use that new API.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a block hierarchy of collection filters and this experimental stock filter block
2. in block attributes set `showCounts` to true and to false to see the behaviours that include/exclude the counts when displaying the filter options
3. Set `displayStyle` to `dropdown` and see that selecting an item always closes the dropdown now
4. Also see that you can click on a selected item in the dropdown to deselect it.


## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


>
